dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1069,6 +1137,51 @@ private static ChatRequestMessage GetRequestMessage(ChatResponseMessage message)
         throw new NotSupportedException($"Role {message.Role} is not supported.");
     }
 
+    private OpenAIChatMessageContent GetChatMessage(ChatChoice chatChoice, ChatCompletions responseData)
+    {
+        var message = new OpenAIChatMessageContent(chatChoice.Message, this.DeploymentOrModelName, GetChatChoiceMetadata(responseData, chatChoice));
+
+        foreach (var toolCall in chatChoice.Message.ToolCalls)
```

Comment:
Just double-checking if `Message` or `ToolCalls` can be null here. They are defined as non-nullable, but sometimes such properties can still hold `null` values. We had similar case with `ChatCompletions.Usage` property, which was fixed in this PR: https://github.com/microsoft/semantic-kernel/pull/4956

Reply:
Good point @dmytrostruk. I quickly scanned the code and found a few places where the collection is used as is, relying solely on the fact that it is declared as non-nullable. The code has been there since the December SK release. Example - https://github.com/microsoft/semantic-kernel/blob/a27a46dc4af47b33df933126e1d5938e919af04c/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIChatMessageContent.cs#L29C156-L29C177